lib/json2xls.js

```diff
@@ -42,55 +47,59 @@ function getByString(object, path) {
 
 //prepare json to be in the correct format for excel-export
 transform.prepareJson = function(json,config) {
-    var res = {};
-    var conf = config||{};
-    var jsonArr = [].concat(json);
-    var fields = conf.fields || Object.keys(jsonArr[0]||{});
-    var types = [];
-    if (!(fields instanceof Array)) {
-        types = Object.keys(fields).map(function(key) {
-            return fields[key];
-        });
-        fields = Object.keys(fields);
-    }
-    //cols
-    res.cols = fields.map(function(key,i) {
-        return {
-            caption: key,
-            type: getType(jsonArr[0][key],types[i]),
-            beforeCellWrite: function(row, cellData, eOpt){
-                eOpt.cellType = getType(cellData,types[i]);
-                return cellData;
-            }
-        };
+  var res = {};
+  var conf = config||{};
+  var jsonArr = [].concat(json);
+  var fields = conf.fields || Object.keys(jsonArr[0]||{});
+  var types = [];
+  var headers = [];
+  if (!(fields instanceof Array)) {
+    types = Object.keys(fields).map(function(key) {
+      return fields[key]['type'] !== undefined ? fields[key]['type'] : fields[key];
     });
-    //rows
-    res.rows = jsonArr.map(function(row) {
-        return fields.map(function(key) {
-            var value = getByString(row,key);
-            //stringify objects
-            if(value && value.constructor == Object) value = JSON.stringify(value);
-            //replace illegal xml characters with a square
-            //see http://www.w3.org/TR/xml/#charsets
-            //#x9 | #xA | #xD | [#x20-#xD7FF] | [#xE000-#xFFFD] | [#x10000-#x10FFFF]
-            if (typeof value === 'string') {
-                value = value.replace(/[^\u0009\u000A\u000D\u0020-\uD7FF\uE000-\uFFFD\u10000-\u10FFFF]/g,'');
-            }
-            return value;
-        });
+    headers = Object.keys(fields).map(function(key) {
+      return fields[key]['header'] !== undefined ? fields[key]['header'] : key;
     });
-    //add style xml if given
-    if (conf.style) {
-        res.stylesXmlFile = conf.style;
-    }
-    return res;
+    fields = Object.keys(fields);
+  }
+  //cols
+  res.cols = fields.map(function(key,i) {
+    return {
+      caption: headers[i],

```

Comment:
When `config` is `undefined`, `fields instanceof Array === true`. 
So it would not go in to the if case telling in **line 56** and `headers` remain `[]`.

When `json` is not empty(`[] || {}`), `fields` is also not empty, then `headers` would not match `fields`.
`headers[i]` in **line 68** would be `undefined`, resulting to the following failure of test.

![image](https://user-images.githubusercontent.com/11766057/50526045-67cc2c80-0b1a-11e9-8d10-c62ae8cfae29.png)

**## fix it**
Replace `headers[i]` to `headers[i] !== undefined ? headers[i] : key` in **line 68**.